iothub/service/src/MessageSystemPropertyNames.cs

```diff
@@ -8,40 +8,122 @@ namespace Microsoft.Azure.Devices
 
+        /// <summary>
+        /// Number of times the message has been previously delivered.
+        /// </summary>
         public const string DeliveryCount = "iothub-deliverycount";
 
+        /// <summary>
+        /// Date and time when the message was received by the server in UTC.
+        /// </summary>
         public const string EnqueuedTime = "iothub-enqueuedtime";
 
+        /// <summary>
+        /// Date and time of message expiration in UTC.
+        /// </summary>
         public const string ExpiryTimeUtc = "absolute-expiry-time";
 
+        /// <summary>
+        /// A string property in a response message that typically contains the MessageId of the request, in request-reply patterns.
+        /// </summary>
         public const string CorrelationId = "correlation-id";
 
+        /// <summary>
+        /// An Id used to specify the origin of messages. When messages are generated by IoT Hub, it is set to name of the IoT hub.
+        /// </summary>
         public const string UserId = "user-id";
 
+        /// <summary>
+        /// IoT hub operation
+        /// </summary>
         public const string Operation = "iothub-operation";
 
+        /// <summary>
+        /// A feedback message generator. This property is used in cloud-to-device messages to request IoT hub to generate feedback messages as a result of the consumption of the message by the device.
+        /// </summary>
+        /// <remarks>
+        /// Possible values:
+        /// <list type="bullet">
+        /// <item>
+        /// <description>none (default): no feedback message is generated.</description>
+        /// </item>
+        /// <item>
+        /// <description>positive: receive a feedback message if the message was completed.</description>
+        /// </item>
+        /// <item>
+        /// <description>negative: receive a feedback message if the message expired (or maximum delivery count was reached) without being completed by the device.</description>
+        /// </item>
+        /// <item>
+        /// <description>full: both positive and negative.</description>
+        /// </item>
+        /// </list>
+        /// </remarks>
         public const string Ack = "iothub-ack";
 
+        /// <summary>
+        /// Specifies the device Id from which this message was sent, if there is one.
+        /// </summary>
         public const string ConnectionDeviceId = "iothub-connection-device-id";
 
+        /// <summary>
+        /// The device generation Id of the target device of the cloud-to-device message.
+        /// DeviceGenerationId is an IoT hub generated, case-sensitive string. This value is used to distinguish devices with the
+        /// same device Id when they have been deleted and re-created.
+        /// </summary>
         public const string ConnectionDeviceGenerationId = "iothub-connection-auth-generation-id";
 
+        /// <summary>
+        /// Connection authentication information.
```

Comment:
'Information' is pretty vague. Is it all the auth data, or just the method being used? IOW, is there sensitive data in here?

Reply:
This is all the data. From tests in the service code I could only find one so I could not add more details. This property contains things like scope, type, issuer etc.. But I don't know what other things you could send or what other auth is possible.

Reply:
Can you put a breakpoint in a test to see what the data looks like?

Reply:
I don't know who sets this to begin with. I will try to play with this in a sample. For now I will remove the xml comments for this so the warning does not go away and address this in my next PR.